### PR TITLE
Surfacing mcf required filters in the backend 

### PR DESCRIPTION
**Pull request by THOR300**
# Description

This pull request surfaces changes made to the `cpr_sdk` to support filtering for the `MCF` corpus data. 

- Rename a data access legacy naming convension. 
- Add family geographies to the `SearchResponseFamily` object. 
- Update the default behaviour for which fields we filter on in vespa from `geography` to `geographies`.
- Bump the `cpr_sdk` version to contain the new queries. 
- Update the test setup so we have the option to use deterministic metadata. This is to enable the metadata test. 
- Update tests. 

### Note:
We can't test for the `corpus_type_name` or `corpus_import_id` in the search response as it isn't included in the object. 

Do we want this as part of this PR or as part of future work? 
- This would mean accessing it in the hit from vespa and adding it to the search response object. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [ ] Patch
- [x] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/schemas/search.py [hunk 1]
```diff
@@ -251,6 +251,11 @@ class SearchResponseFamily(BaseModel):
     The geographical location of the family in ISO 3166-1 alpha-3
     """
 
+    family_geographies: List[str]
+    """
+    The geographical locations of the family in ISO 3166-1 alpha-3
```
Comment (jesse-c) [inline app/api/api_v1/schemas/search.py:256]: **suggestion:**
```suggestion
    The geographical locations of the families in ISO 3166-1 alpha-3
```

File: app/core/search.py [hunk 2]
```diff
@@ -391,6 +391,7 @@ def _process_vespa_search_response_families(
                 or hit.family_category is None
                 or hit.family_source is None
                 or hit.family_geography is None
```
Comment (jesse-c) [inline app/core/search.py:393]: **question:** If this was replaced above, could it be removed from here?

**Review by jesse-c — COMMENTED** [review]
Looks reasoanble to me. I don't think I have enough understanding of this service though!

File: app/api/api_v1/schemas/search.py (hunk 1 above)
Comment (THOR300) [reply app/api/api_v1/schemas/search.py:256]: So in this context it's we have a search response for a family and thus singular!

**Review by olaughter — COMMENTED** [review]
Functionally this looks great to me, and I think the tests you've added look good. I am wondering if we might need one more test in the search tests that makes sure we get metadata back? Like how we are currently doing for geographies. This would give us good confidence in this all fitting together correctly. This is good to go from my view otherwise!

**Comment by THOR300** [comment]
A thought - add multiple geographies to test data to ensure we're ready for that as I think they're all just a list of the single geo currently.

**Comment by THOR300** [comment]
Have you spun up the backend locally outside of tests and ran a query?

**Comment by THOR300** [comment]
> Functionally this looks great to me, and I think the tests you've added look good. I am wondering if we might need one more test in the search tests that makes sure we get metadata back? Like how we are currently doing for geographies. This would give us good confidence in this all fitting together correctly. This is good to go from my view otherwise!

Definitely!

File: app/core/search.py (hunk 2 above)
Comment (THOR300) [reply app/core/search.py:393]: I'm not sure I follow! This is a check as to whether the `hit` that is returned from vespa has all the relevant information to construct a `SearchResponseFamily` object.

File: tests/search/setup_search_tests.py [hunk 3]
```diff
@@ -75,7 +78,10 @@ def _populate_db_families(db: Session, max_docs: int = VESPA_FIXTURE_COUNT) -> N
         if doc["fields"]["family_document_ref"] not in seen_family_ids:
             _create_family(db, family)
             _create_family_event(db, family)
-            _create_family_metadata(db, family)
+            if not deterministic_metadata:
```
Comment (olaughter) [inline tests/search/setup_search_tests.py:81]: Do tests fail if we just make all of them deterministic?

File: tests/search/setup_search_tests.py [hunk 4]
```diff
@@ -182,6 +188,24 @@ def _create_family_metadata(db: Session, family: VespaFixture):
...
 
 
+def _create_family_metadata_deterministic(db: Session, family: VespaFixture):
+    metadata_values = defaultdict(list)
+    family_metadata = family["fields"]["metadata"]
+    if not family_metadata:
+        return
+    for metadata in family_metadata:
+        name = metadata["name"]  # type: ignore
+        value = metadata["value"]  # type: ignore
```
Comment (olaughter) [inline tests/search/setup_search_tests.py:198]: Wondering what the linting issue is that is being ignored here? If we definitely need it it'd be good to add the reason

Comment (olaughter) [reply tests/search/setup_search_tests.py:198]: suggestion: Not sure if this makes a linting difference, but super minor simplification if you are interested
```suggestion
        name, value = metadata.values()
```

**Review by olaughter — APPROVED** [review]
Looking good! Some minor points to make decisions on below. Nothing blocking though, good work! 👍 👍

File: tests/search/setup_search_tests.py (hunk 4 above)
Comment (THOR300) [reply tests/search/setup_search_tests.py:198]: image

For some reason the linter thinks metadata is a string

File: tests/search/setup_search_tests.py (hunk 3 above)
Comment (THOR300) [reply tests/search/setup_search_tests.py:81]: Yes there was like ~9 failures so given the time I just went with this approach!

Comment (olaughter) [reply tests/search/setup_search_tests.py:81]: Sounds reasonable 👍

File: tests/search/setup_search_tests.py (hunk 4 above)
Comment (olaughter) [reply tests/search/setup_search_tests.py:198]: Ah this is probably a failing of `VespaFixture`

Comment (olaughter) [reply tests/search/setup_search_tests.py:198]: Recommend leaving it then in that case, we can resolve it another time

**Pull request merged**